core/trino-main/src/test/java/io/trino/sql/planner/TestLogicalPlanner.java

```diff
@@ -2631,6 +2632,31 @@ public void testPruneWindow()
                         values("c")));
     }
 
+    @Test
+    public void testCrossJoinUnnest()
+    {
+        assertPlan(
+                """
+                        SELECT *
```

Comment:
I just noticed that the outer select is redundant. You can drop it to simplify the query to
```
WITH RECURSIVE recursive_call (level) AS (
                                SELECT 1 AS level
                                FROM (SELECT ARRAY[] AS array) AS t1 CROSS JOIN UNNEST(array) AS t2 (x)
                                UNION ALL
                                SELECT recursive_call.level + 1 AS level
                                FROM recursive_call INNER JOIN nation AS c ON recursive_call.level <= 1
                            )
                            SELECT * FROM recursive_call
```